### PR TITLE
Refactor endpoints

### PR DESCRIPTION
**Pull request by rossjones**
Refactors the endpoints to change the behaviour of the system.  Before this PR each endpoint handled the GET and POST requests for each step, rendering the template, processing the state, and then redirecting to the next step.  Although this worked, it did not empower users of the plugin to use it how they want to.

After this PR, the plugin is composed of:

* Macros that will render specific parts of the upload state.  e.g. select a sheet, or review the data etc.
* Template functions that can be used in forms to specify how to process the current step, and then the URL to redirect to afterwards
* Endpoints to handle POST requests from forms and then redirect to the requested URL.
* Templates showing how the macros can be used either for education, or to be copied into the users' prototype.

### CODE REVIEW

File: lib/importer/index.js [hunk 1]
```diff
@@ -2,141 +2,221 @@ const multer = require("multer");
...
+const IMPORTER_SESSION_KEY = "importer.session"
+const IMPORTER_ERROR_KEY = "importer.error"
+
+//--------------------------------------------------------------------
+// The endpoints used to receive POST requests from importer macros, 
+// and used to generate template functions that can be used to find 
+// the URL for each step (to be used in form actions)
+//--------------------------------------------------------------------
+const IMPORTER_ROUTE_MAP = new Map([
+  ["importerStartPath", "/importer/start"],
```
Comment (alaric-rd) [inline lib/importer/index.js:14]: I'm nervous that "importer" might be too general a name to inject into app-global namespaces. People never have to type these by hand, so would something like "spreadsheet-importer" be fine? Or am I worrying too much?

File: lib/importer/index.js [hunk 2]
```diff
@@ -2,141 +2,221 @@ const multer = require("multer");
...
+  // allow users to find the path that they should use to submit data, 
+  // and also allows them to specify which url to redirect to after 
+  // the POST data has been processed.
+  //--------------------------------------------------------------------  
+  for ([key, value] of IMPORTER_ROUTE_MAP) {
+    const k = key;
+    const v = value;
+
+    prototypeKit.views.addFunction(k, (next)=>{
+      return `${v}?next=${next}`;
```
Comment (alaric-rd) [inline lib/importer/index.js:43]: We don't need any URL-encoding of the next param here, right? ...right?

**Review by alaric-rd — APPROVED** [review]
LGTM, although I'm not familiar with all the nunjucks stuff so just glossed over that

File: lib/importer/index.js (hunk 1 above)
Comment (rossjones) [reply lib/importer/index.js:14]: I think I agree, I just can't think of a better name to use and we're annoyingly constrained in the patterns we can use for namespacing (e.g no -). I'll see if I can get concensus on naming and then submit a PR with names changed.

File: lib/importer/index.js (hunk 2 above)
Comment (rossjones) [reply lib/importer/index.js:43]: ¯\_(ツ)_/¯ but I'll change it anyway to be safe :D

**Pull request merged**